backend/database.py

```diff
@@ -1,26 +1,226 @@
-from flask import Flask
-from flask_sqlalchemy import SQLAlchemy
-
-app = Flask(__name__)
-app.config["SQLALCHEMY_DATABASE_URI"] = "sqlite:///lab8.sqlite" 
-db = SQLAlchemy(app)
-
-class User(db.Model):
-    u_name = db.Column(db.String, primary_key=True)
-    u_password = db.Column(db.String, nullable=False)
-    u_type = db.Column(db.String, nullable=False)
-
-class Classes(db.Model):
-    c_classkey = db.Column(db.Integer, nullable=False, primary_key=True)
-    c_classname = db.Column(db.String, nullable=False)
-    c_teacher = db.Column(db.String, nullable=False)
-    c_time = db.Column(db.String, nullable=False)
-    c_seats = db.Column(db.Integer, nullable=False)
-
-class Students_in_Classes(db.Model):
-    sc_classkey = db.Column(db.Integer, nullable=False, primary_key=True)
-    sc_name = db.Column(db.String, nullable=False)
-    sc_grade = db.Column(db.Integer, nullable=False)
-
-db.create_all()
+from ast import get_docstring
+import sqlite3
+
+def get_all_of_role(c, role):
+    command = '''
+    SELECT * FROM Users
+    WHERE u_role = \"''' + str(role) +'''\" ;
+    '''
+
+    cur = c.cursor()
+    cur.execute(command)
+    result = cur.fetchall()
+    print(result)
+    return result
+
+def get_all_classes(c):
+    command = '''
+    SELECT * FROM Classes
+    '''
+    cur = c.cursor()
+    cur.execute(command)
+    result = cur.fetchall()
+    print(result)
+    return result
+
+def get_all_users(c):
+    command = '''
+    SELECT * FROM Users
+    '''
+    cur = c.cursor()
+    cur.execute(command)
+    result = cur.fetchall()
+    print(result)
+    return result
+
+def get_user(user, c):
+    command = '''
+    SELECT * FROM Users WHERE u_name =''' + "\""+ user +"\"" +";"
+    cur = c.cursor()
+    cur.execute(command)
+    result = cur.fetchone()
+    return result
+
+def get_classes_of_student(c, user):
+    command = '''
+    SELECT c_classname, c_teacher, c_time, c_seats
+    FROM Classes, Students_in_Classes
+    WHERE sc_name = \"''' + user + '''\"
+    and sc_classkey = c_classkey
+    '''
+    cur = c.cursor()
+    cur.execute(command)
+    result = cur.fetchall()
+    print(result)
+    return result
+
+def get_classes_of_teacher(c, user):
+    command = '''
+    SELECT c_classkey, c_classname, c_teacher, c_time, c_seats
+    FROM Classes
+    WHERE c_teacher = \"''' + user + '''\"'''
+    cur = c.cursor()
+    cur.execute(command)
+    result = cur.fetchall()
+    print(result)
+    return result
+
+def get_students_in_classes(c):
+    command = '''
+    SELECT * FROM Students_in_CLasses
+    '''
+    cur = c.cursor()
+    cur.execute(command)
+    result = cur.fetchall()
+    print(result)
+    return result
+
+def get_students_in_a_class(classkey, c):
+    command = '''
+    SELECT sc_name, sc_grade
+    FROM Students_in_Classes
+    WHERE sc_classkey = ''' + str(classkey)
+    cur = c.cursor()
+    cur.execute(command)
+    result = cur.fetchall()
+    print(result)
+    return result
+
+def create_db(c):
```

Comment:
PEP8 says we should be using verbose variable names. Single character variable names are awful because you can't even find/replace them; you have to use a semantic tool.